### ISSUE TITLE
Overstating rainfall

### ISSUE THREAD

**Issue by ianatkin80** [matthewwall/weewx-netatmo#1]
The rainfall total reported by weewx via weewx-netatmo is much higher than through the netatmo app.  As an example, at one point today netatmo was reporting 1.0mm and weewx 3.0mm.

Could a cm to inches conversion be at fault?

**Comment by matthewwall** [comment]
it is quite possible that i got the units reversed - the netatmo documentation is not clear, and i do not have a rain unit on which to test.

could you do the following test:

1) stop weewx
2) start weewx directly (not as daemon): 'sudo weewxd /etc/weewx/weewx.conf'
3) put water in the rain bucket until you get one bucket tip
4) make a note of the LOOP and REC produced by weewx
5) make a note of the rain amount reported by the netatmo web site/app

thanks for your help with this!

**Comment by ianatkin80** [comment]
One bucket tip produced no output on LOOP and REC so I poured 100ml through
- the app reports 3.0mm and this is the LOOP output:

LOOP:   2016-07-05 19:04:24 BST (1467741864) altimeter: 30.1067074365,
appTemp: 61.1434535318, barometer: 30.1152026453, battery_vp: 6345,
cloudbase: 3261.61138313, co2: 1306, dateTime: 1467741864, dewpoint:
48.1776238254, heatindex: 62.24, humidex: 63.7247075415, inDewpoint:
60.134214445, inHumidity: 71, inTemp: 69.98, maxSolarRad: None, noise: 35,
outHumidity: 60, outTemp: 62.24, pressure: 30.0443000591, rain:
0.119291338461, rainRate: 0.477165353844, rf_status: 66, usUnits: 1,
wifi_status: 51, windchill: 62.24, windDir: 318, windGust: 6.21371192,
windGustDir: 308, windSpeed: 1.242742384

**Comment by matthewwall** [comment]
the LOOP value of 0.11929 is inches (usUnits=1), which is the same as 3 mm.

do you see a value of 3 mm in the weewx report?

**Comment by ianatkin80** [comment]
In the weewx report it looks like ~9.5mm

**Comment by matthewwall** [comment]
this does not make sense.

if the LOOP value is 0.11929 inches, and the amount reported by the app is 0.3 mm, then there is no problem - the values match.

so that leaves a discrepancy between the 0.11929 inches in the weewx LOOP, and a value of 9.5 mm in a weewx report.  where is the 9.5 mm value coming from?  is that a value displayed in a template, e.g., $current.rain?  or is it actually a sum, such as $day.rain.sum?  or is it something else?

**Comment by ianatkin80** [comment]
I can't make sense of it either. App is reporting 7mm total rain today and
weewx is reporting 31.6mm. The same value has been passed to Weather
Underground so it's not a template issue?
On 8 Jul 2016 2:43 a.m., "Matthew Wall" notifications@github.com wrote:

this does not make sense.

if the LOOP value is 0.11929 inches, and the amount reported by the app is
0.3 mm, then there is no problem - the values match.

so that leaves a discrepancy between the 0.11929 inches in the weewx LOOP,
and a value of 9.5 mm in a weewx report. where is the 9.5 mm value coming
from? is that a value displayed in a template, e.g., $current.rain? or is
it actually a sum, such as $day.rain.sum? or is it something else?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
https://github.com/matthewwall/weewx-netatmo/issues/1#issuecomment-231255326,
or mute the thread
https://github.com/notifications/unsubscribe/AJ4mW9vAzveYBBMIhFTMVmRBdhqLuAEbks5qTasngaJpZM4JEk11
.

**Comment by matthewwall** [comment]
we need to determine exactly what weewx does for a single rain event.  when you posted the single LOOP packet, was there any rain reported in the LOOP packets before or after the one you posted?

try another test while running weewx directly, but this time record weewx output for every LOOP and REC from about 10 minutes before you do the water/buckettip until 10 minutes after you do the water/buckettip.

**Comment by ianatkin80** [comment]
Sorry it's taken so long, I was ill then it's been very wet here!  This is the output:

```
LOOP:   2016-07-23 20:12:55 BST (1469301175) altimeter: 30.0682402964, appTemp: 70.8226918085, barometer: 30.0757777394, battery_vp: 6117, cloudbase: 3053.11760301, co2: 421, dateTime: 1469301175, dewpoint: 56.1149964579, heatindex: 69.26, humidex: 74.6928772989, inDewpoint: 57.763931153, inHumidity: 61, inTemp: 71.96, maxSolarRad: None, noise: 55, outHumidity: 63, outTemp: 69.26, pressure: 30.0059066745, rain: 0.0, rainRate: 0, rf_status: 69, usUnits: 1, wifi_status: 49, windchill: 69.26, windDir: 310, windGust: 4.970969536, windGustDir: 287, windSpeed: 0.621371192
LOOP:   2016-07-23 20:22:58 BST (1469301778) altimeter: 30.0711993078, appTemp: 70.257964473, barometer: 30.0788094503, battery_vp: 6117, cloudbase: 2947.46021234, co2: 408, dateTime: 1469301778, dewpoint: 56.0398889769, heatindex: 68.72, humidex: 74.1104425437, inDewpoint: 57.135185085, inHumidity: 60, inTemp: 71.78, maxSolarRad: None, noise: 35, outHumidity: 64, outTemp: 68.72, pressure: 30.0088600118, rain: 0.0, rainRate: 0, rf_status: 69, usUnits: 1, wifi_status: 47, windchill: 68.72, windDir: 315, windGust: 2.485484768, windGustDir: 315, windSpeed: 0.621371192
REC:    2016-07-23 20:15:00 BST (1469301300) altimeter: 30.0682402964, appTemp: 70.8226918085, barometer: 30.0757777394, battery_vp: 6117.0, cloudbase: 3053.11760301, co2: 421.0, dateTime: 1469301300.0, dewpoint: 56.1149964579, ET: None, heatindex: 69.26, humidex: 74.6928772989, inDewpoint: 57.763931153, inHumidity: 61.0, inTemp: 71.96, interval: 5, maxSolarRad: None, noise: 55.0, outHumidity: 63.0, outTemp: 69.26, pressure: 30.0059066745, rain: 0.0, rainRate: 0.0, rf_status: 69.0, usUnits: 1, wifi_status: 49.0, windchill: 69.26, windDir: 310.0, windGust: 4.970969536, windGustDir: 287, windrun: 4.55672207467, windSpeed: 0.621371192
LOOP:   2016-07-23 20:32:57 BST (1469302377) altimeter: 30.0711993078, appTemp: 70.257964473, barometer: 30.0788094503, battery_vp: 6117, cloudbase: 2947.46021234, co2: 408, dateTime: 1469302377, dewpoint: 56.0398889769, heatindex: 68.72, humidex: 74.1104425437, inDewpoint: 57.135185085, inHumidity: 60, inTemp: 71.78, maxSolarRad: None, noise: 35, outHumidity: 64, outTemp: 68.72, pressure: 30.0088600118, rain: 0.0, rainRate: 0, rf_status: 69, usUnits: 1, wifi_status: 47, windchill: 68.72, windDir: 315, windGust: 2.485484768, windGustDir: 315, windSpeed: 0.621371192
REC:    2016-07-23 20:25:00 BST (1469301900) altimeter: 30.0711993078, appTemp: 70.257964473, barometer: 30.0788094503, battery_vp: 6117.0, cloudbase: 2947.46021234, co2: 408.0, dateTime: 1469301900.0, dewpoint: 56.0398889769, ET: None, heatindex: 68.72, humidex: 74.1104425437, inDewpoint: 57.135185085, inHumidity: 60.0, inTemp: 71.78, interval: 5, maxSolarRad: None, noise: 35.0, outHumidity: 64.0, outTemp: 68.72, pressure: 30.0088600118, rain: 0.0, rainRate: 0.0, rf_status: 69.0, usUnits: 1, wifi_status: 47.0, windchill: 68.72, windDir: 315.0, windGust: 2.485484768, windGustDir: 315, windrun: 4.60850300733, windSpeed: 0.621371192
LOOP:   2016-07-23 20:42:57 BST (1469302977) altimeter: 30.0741583191, appTemp: 69.5934246054, barometer: 30.0818652376, battery_vp: 6121, cloudbase: 2745.59493905, co2: 380, dateTime: 1469302977, dewpoint: 56.2080961794, heatindex: 68.0, humidex: 73.4856226514, inDewpoint: 57.2561888848, inHumidity: 61, inTemp: 71.42, maxSolarRad: None, noise: 44, outHumidity: 66, outTemp: 68.0, pressure: 30.0118133491, rain: 1.07944105113, rainRate: 4.3177642045, rf_status: 68, usUnits: 1, wifi_status: 50, windchill: 68.0, windDir: 270, windGust: 3.10685596, windGustDir: 288, windSpeed: 0.621371192
REC:    2016-07-23 20:35:00 BST (1469302500) altimeter: 30.0711993078, appTemp: 70.257964473, barometer: 30.0788094503, battery_vp: 6117.0, cloudbase: 2947.46021234, co2: 408.0, dateTime: 1469302500.0, dewpoint: 56.0398889769, ET: None, heatindex: 68.72, humidex: 74.1104425437, inDewpoint: 57.135185085, inHumidity: 60.0, inTemp: 71.78, interval: 5, maxSolarRad: None, noise: 35.0, outHumidity: 64.0, outTemp: 68.72, pressure: 30.0088600118, rain: 0.0, rainRate: 0.0, rf_status: 69.0, usUnits: 1, wifi_status: 47.0, windchill: 68.72, windDir: 315.0, windGust: 2.485484768, windGustDir: 315, windrun: 4.66028394, windSpeed: 0.621371192
LOOP:   2016-07-23 20:53:00 BST (1469303580) altimeter: 30.0771173303, appTemp: 69.1209939323, barometer: 30.0848733541, battery_vp: 6121, cloudbase: 2741.48950458, co2: 398, dateTime: 1469303580, dewpoint: 55.8661600911, heatindex: 67.64, humidex: 72.9326875311, inDewpoint: 57.0869335573, inHumidity: 61, inTemp: 71.24, maxSolarRad: None, noise: 48, outHumidity: 66, outTemp: 67.64, pressure: 30.0147666864, rain: 0.0, rainRate: 4.3177642045, rf_status: 69, usUnits: 1, wifi_status: 47, windchill: 67.64, windDir: 305, windGust: 4.349598344, windGustDir: 297, windSpeed: 0.621371192
REC:    2016-07-23 20:45:00 BST (1469303100) altimeter: 30.0741583191, appTemp: 69.5934246054, barometer: 30.0818652376, battery_vp: 6121.0, cloudbase: 2745.59493905, co2: 380.0, dateTime: 1469303100.0, dewpoint: 56.2080961794, ET: None, heatindex: 68.0, humidex: 73.4856226514, inDewpoint: 57.2561888848, inHumidity: 61.0, inTemp: 71.42, interval: 5, maxSolarRad: None, noise: 44.0, outHumidity: 66.0, outTemp: 68.0, pressure: 30.0118133491, rain: 1.07944105113, rainRate: 4.3177642045, rf_status: 68.0, usUnits: 1, wifi_status: 50.0, windchill: 68.0, windDir: 270.0, windGust: 3.10685596, windGustDir: 288, windrun: 4.71206487267, windSpeed: 0.621371192
LOOP:   2016-07-23 21:03:00 BST (1469304180) altimeter: 30.0771173303, appTemp: 69.1209939323, barometer: 30.0848733541, battery_vp: 6121, cloudbase: 2741.48950458, co2: 398, dateTime: 1469304180, dewpoint: 55.8661600911, heatindex: 67.64, humidex: 72.9326875311, inDewpoint: 57.0869335573, inHumidity: 61, inTemp: 71.24, maxSolarRad: None, noise: 48, outHumidity: 66, outTemp: 67.64, pressure: 30.0147666864, rain: 0.0, rainRate: 0, rf_status: 69, usUnits: 1, wifi_status: 47, windchill: 67.64, windDir: 305, windGust: 4.349598344, windGustDir: 297, windSpeed: 0.621371192
REC:    2016-07-23 20:55:00 BST (1469303700) altimeter: 30.0771173303, appTemp: 69.1209939323, barometer: 30.0848733541, battery_vp: 6121.0, cloudbase: 2741.48950458, co2: 398.0, dateTime: 1469303700.0, dewpoint: 55.8661600911, ET: None, heatindex: 67.64, humidex: 72.9326875311, inDewpoint: 57.0869335573, inHumidity: 61.0, inTemp: 71.24, interval: 5, maxSolarRad: None, noise: 48.0, outHumidity: 66.0, outTemp: 67.64, pressure: 30.0147666864, rain: 0.0, rainRate: 4.3177642045, rf_status: 69.0, usUnits: 1, wifi_status: 47.0, windchill: 67.64, windDir: 305.0, windGust: 4.349598344, windGustDir: 297, windrun: 4.76384580533, windSpeed: 0.621371192
```

Attached is the netatmo app result, my current rain gauge calibration and what weewx output.
![screenshot_20160723-203936](https://cloud.githubusercontent.com/assets/10364507/17080000/1b67d8f2-5119-11e6-80ab-687af2e341c6.png)
![screenshot_20160723-204018](https://cloud.githubusercontent.com/assets/10364507/17080001/1b6a4ad8-5119-11e6-8b35-f16a252ada47.png)
![screenshot_20160723-205844](https://cloud.githubusercontent.com/assets/10364507/17080002/1b6acf62-5119-11e6-9a55-59d5ea999189.png)

**Comment by ianatkin80** [comment]
This page may give you better guidance than I can!
[https://dev.netatmo.com/dev/resources/technical/reference/weatherstation/getstationsdata](url)
It has the following return parameters for the rain gauge:

```
Rain (mm)
sum_rain_1: rain measured for the last hour (mm)
sum_rain_24: rain measured for past 24h(mm)
time_utc: timestamp when data was measured
```

**Comment by ianatkin80** [comment]
The above has now been confirmed by netatmo: http://forum.netatmo.com/viewtopic.php?f=5&t=6596

**Comment by matthewwall** [comment]
ian, please try version 0.3 of the driver.  do a single bucket tip and make a sure that the LOOP, REC, and app values agree.  thank you!

**Comment by ianatkin80** [comment]
I can't see v0.3, can you make sure it has published please

**Comment by matthewwall** [comment]
v0.3rc1 is now a tagged release

**Comment by ianatkin80** [comment]
No rain has shown in LOOP or REC at all, app is reporting 0.4mm:

```
LOOP:   2016-08-06 14:08:13 BST (1470488893) altimeter: 30.2309857665, appTemp: 78.9281611714, barometer: 30.2374360038, battery_vp: 5985, cloudbase: 5313.53473847, co2: 951, dateTime: 1470488893, dewpoint: 54.9891610619, heatindex: 78.08, humidex: 82.8876373098, inDewpoint: 57.9844373378, inHumidity: 63, inTemp: 71.24, maxSolarRad: None, noise: 50, outHumidity: 45, outTemp: 78.08, pressure: 30.1683402245, rain: 0.0, rainRate: 0, rf_status: 76, usUnits: 1, wifi_status: 48, windchill: 78.08, windDir: 296, windGust: 5.592340728, windGustDir: 144, windSpeed: 1.242742384
LOOP:   2016-08-06 14:18:14 BST (1470489494) altimeter: 30.2309857665, appTemp: 78.9281611714, barometer: 30.2374360038, battery_vp: 5985, cloudbase: 5313.53473847, co2: 951, dateTime: 1470489494, dewpoint: 54.9891610619, heatindex: 78.08, humidex: 82.8876373098, inDewpoint: 57.9844373378, inHumidity: 63, inTemp: 71.24, maxSolarRad: None, noise: 50, outHumidity: 45, outTemp: 78.08, pressure: 30.1683402245, rain: 0.0, rainRate: 0, rf_status: 76, usUnits: 1, wifi_status: 48, windchill: 78.08, windDir: 296, windGust: 5.592340728, windGustDir: 144, windSpeed: 1.242742384
REC:    2016-08-06 14:10:00 BST (1470489000) altimeter: 30.2309857665, appTemp: 78.9281611714, barometer: 30.2374360038, battery_vp: 5985.0, cloudbase: 5313.53473847, co2: 951.0, dateTime: 1470489000.0, dewpoint: 54.9891610619, ET: None, heatindex: 78.08, humidex: 82.8876373098, inDewpoint: 57.9844373378, inHumidity: 63.0, inTemp: 71.24, interval: 5, maxSolarRad: None, noise: 50.0, outHumidity: 45.0, outTemp: 78.08, pressure: 30.1683402245, rain: 0.0, rainRate: 0, rf_status: 76.0, usUnits: 1, wifi_status: 48.0, windchill: 78.08, windDir: 296.0, windGust: 5.592340728, windGustDir: 144, windrun: 1.55342798, windSpeed: 1.242742384
LOOP:   2016-08-06 14:28:15 BST (1470490095) altimeter: 30.2280267608, appTemp: 78.8493190529, barometer: 30.2345222384, battery_vp: 5987, cloudbase: 5033.56329182, co2: 895, dateTime: 1470490095, dewpoint: 55.8610354272, heatindex: 77.72, humidex: 83.0098123995, inDewpoint: 57.7084537485, inHumidity: 62, inTemp: 71.42, maxSolarRad: None, noise: 49, outHumidity: 47, outTemp: 77.72, pressure: 30.1653868872, rain: 0.0, rainRate: 0, rf_status: 77, usUnits: 1, wifi_status: 49, windchill: 77.72, windDir: 135, windGust: 4.970969536, windGustDir: 82, windSpeed: 1.242742384
REC:    2016-08-06 14:20:00 BST (1470489600) altimeter: 30.2309857665, appTemp: 78.9281611714, barometer: 30.2374360038, battery_vp: 5985.0, cloudbase: 5313.53473847, co2: 951.0, dateTime: 1470489600.0, dewpoint: 54.9891610619, ET: None, heatindex: 78.08, humidex: 82.8876373098, inDewpoint: 57.9844373378, inHumidity: 63.0, inTemp: 71.24, interval: 5, maxSolarRad: None, noise: 50.0, outHumidity: 45.0, outTemp: 78.08, pressure: 30.1683402245, rain: 0.0, rainRate: 0, rf_status: 76.0, usUnits: 1, wifi_status: 48.0, windchill: 78.08, windDir: 296.0, windGust: 5.592340728, windGustDir: 144, windrun: 1.65698984533, windSpeed: 1.242742384
LOOP:   2016-08-06 14:38:16 BST (1470490696) altimeter: 30.2309857665, appTemp: 78.8301407387, barometer: 30.2375752122, battery_vp: 5989, cloudbase: 4630.13407648, co2: 873, dateTime: 1470490696, dewpoint: 56.9161239747, heatindex: 77.0, humidex: 82.8920576048, inDewpoint: 57.7084537485, inHumidity: 62, inTemp: 71.42, maxSolarRad: None, noise: 50, outHumidity: 50, outTemp: 77.0, pressure: 30.1683402245, rain: 0.0, rainRate: 0, rf_status: 120, usUnits: 1, wifi_status: 51, windchill: 77.0, windDir: 161, windGust: 3.10685596, windGustDir: 177, windSpeed: 0.621371192
REC:    2016-08-06 14:30:00 BST (1470490200) altimeter: 30.2280267608, appTemp: 78.8493190529, barometer: 30.2345222384, battery_vp: 5987.0, cloudbase: 5033.56329182, co2: 895.0, dateTime: 1470490200.0, dewpoint: 55.8610354272, ET: None, heatindex: 77.72, humidex: 83.0098123995, inDewpoint: 57.7084537485, inHumidity: 62.0, inTemp: 71.42, interval: 5, maxSolarRad: None, noise: 49.0, outHumidity: 47.0, outTemp: 77.72, pressure: 30.1653868872, rain: 0.0, rainRate: 0, rf_status: 77.0, usUnits: 1, wifi_status: 49.0, windchill: 77.72, windDir: 135.0, windGust: 4.970969536, windGustDir: 82, windrun: 1.76055171067, windSpeed: 1.242742384
LOOP:   2016-08-06 14:48:16 BST (1470491296) altimeter: 30.2309857665, appTemp: 78.8301407387, barometer: 30.2375752122, battery_vp: 5989, cloudbase: 4630.13407648, co2: 873, dateTime: 1470491296, dewpoint: 56.9161239747, heatindex: 77.0, humidex: 82.8920576048, inDewpoint: 57.7084537485, inHumidity: 62, inTemp: 71.42, maxSolarRad: None, noise: 50, outHumidity: 50, outTemp: 77.0, pressure: 30.1683402245, rain: 0.0, rainRate: 0, rf_status: 120, usUnits: 1, wifi_status: 51, windchill: 77.0, windDir: 161, windGust: 3.10685596, windGustDir: 177, windSpeed: 0.621371192
REC:    2016-08-06 14:40:00 BST (1470490800) altimeter: 30.2309857665, appTemp: 78.8301407387, barometer: 30.2375752122, battery_vp: 5989.0, cloudbase: 4630.13407648, co2: 873.0, dateTime: 1470490800.0, dewpoint: 56.9161239747, ET: None, heatindex: 77.0, humidex: 82.8920576048, inDewpoint: 57.7084537485, inHumidity: 62.0, inTemp: 71.42, interval: 5, maxSolarRad: None, noise: 50.0, outHumidity: 50.0, outTemp: 77.0, pressure: 30.1683402245, rain: 0.0, rainRate: 0, rf_status: 120.0, usUnits: 1, wifi_status: 51.0, windchill: 77.0, windDir: 161.0, windGust: 3.10685596, windGustDir: 177, windrun: 1.864113576, windSpeed: 0.621371192
LOOP:   2016-08-06 14:58:18 BST (1470491898) altimeter: 30.2309857665, appTemp: 79.4695153888, barometer: 30.2374591663, battery_vp: 5985, cloudbase: 4774.85038991, co2: 911, dateTime: 1470491898, dewpoint: 57.1793721956, heatindex: 77.9, humidex: 83.9455874361, inDewpoint: 57.8780432221, inHumidity: 62, inTemp: 71.6, maxSolarRad: None, noise: 50, outHumidity: 49, outTemp: 77.9, pressure: 30.1683402245, rain: 0.0, rainRate: 0, rf_status: 75, usUnits: 1, wifi_status: 50, windchill: 77.9, windDir: 149, windGust: 4.970969536, windGustDir: 168, windSpeed: 1.242742384
REC:    2016-08-06 14:50:00 BST (1470491400) altimeter: 30.2309857665, appTemp: 78.8301407387, barometer: 30.2375752122, battery_vp: 5989.0, cloudbase: 4630.13407648, co2: 873.0, dateTime: 1470491400.0, dewpoint: 56.9161239747, ET: None, heatindex: 77.0, humidex: 82.8920576048, inDewpoint: 57.7084537485, inHumidity: 62.0, inTemp: 71.42, interval: 5, maxSolarRad: None, noise: 50.0, outHumidity: 50.0, outTemp: 77.0, pressure: 30.1683402245, rain: 0.0, rainRate: 0, rf_status: 120.0, usUnits: 1, wifi_status: 51.0, windchill: 77.0, windDir: 161.0, windGust: 3.10685596, windGustDir: 177, windrun: 1.91589450867, windSpeed: 0.621371192
```

**Comment by ianatkin80** [comment]
Ok, we had some actual rain overnight and the weewx report shows exactly 2.54x more rain than the app. I'm assuming that's not a coincidence!

**Comment by matthewwall** [comment]
fixed at commit c215b57
